src/components/shared/ButtonWithPanel.js

```diff
@@ -43,6 +44,14 @@ class ButtonWithPanel extends React.PureComponent<Props, State> {
     super(props);
     this.state = { open: !!props.open };
   }
+  // the panel can be closed by pressing the Esc key
+  componentDidMount() {
+    window.addEventListener('keydown', this._onKeyDown);
+  }
```

Comment:
Unfortunately there's a conflict with latest master, as there's already a `componentDidMount` below, that's been added by the other patch for the MenuButtons. Can you please rebase locally and add this change to the existing `componentDidMount`?

Ideally the order of methods should stay `componentDidMount` then `componentWillUnmount`.